Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -7575,7 +7575,12 @@ void WebPage::getTextFragmentMatch(CompletionHandler<void(const String&)>&& comp
         return;
     }
 
-    FragmentDirectiveParser fragmentDirectiveParser(mainWebFrame().url());
+    auto fragmentDirective = document->fragmentDirective();
```

Comment:
Don't we need to make the same change in WebKitLegacy?

Reply:
I'll consider this for a later change.